### PR TITLE
Adding artifacts and the code to build on iOS

### PR DESCRIPTION
**Pull request by gkassabli**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
This PR adds configuration for running UnityGameLogic on OS X, configuration to Unity project for running on iOS and renames android_launch to mobile_launch spatial configuration for consistency
#### Tests
How did you test these changes prior to submitting this pull request?
Run spatial with mobile_launch configuration and iOS device and ensure they work together
#### Documentation
How is this documented (for example: release note, upgrade guide, feature page, in-code documentation)?
#### Primary reviewers
@jessicafalk @DanailPenev

### CODE REVIEW

File: workers/unity/ProjectSettings/ProjectSettings.asset [hunk 1]
```diff
@@ -608,8 +611,8 @@ PlayerSettings:
   webGLCompressionFormat: 1
   webGLLinkerTarget: 0
   scriptingDefineSymbols:
-    1: UNITY_POST_PROCESSING_STACK_V2;UNITY_DISABLE_AUTOMATIC_SYSTEM_BOOTSTRAP
-    4: UNITY_POST_PROCESSING_STACK_V2
+    1: UNITY_POST_PROCESSING_STACK_V2;UNITY_DISABLE_AUTOMATIC_SYSTEM_BOOTSTRAP;IMPROBABLE_WORKERTYPE_UNITYCLIENT
```
Comment (DanailPenev) [inline workers/unity/ProjectSettings/ProjectSettings.asset]: Don't know if we need the IMPROBABLE_WORKERTYPE_UNITYCLIENT (see this https://github.com/spatialos/UnityGDK/pull/247#discussion_r212236953 )

Comment (gkassabli) [reply workers/unity/ProjectSettings/ProjectSettings.asset]: Good point, I'll recheck that

**Comment by jessicafalk** [comment]
Is the ` WorkerCoreStatic.dll` supposed to replace `WorkerCore.dll` ?

**Comment by DanailPenev** [comment]
Also if it's possible (i.e. if Unity does not cry about it), could we have it in an iOS folder in Plugins -> Improbable -> Core (it should be core worker lib anyway?) to stay consistent with other platforms?

File: workers/unity/ProjectSettings/ProjectSettings.asset (hunk 1 above)
Comment (jessicafalk) [reply workers/unity/ProjectSettings/ProjectSettings.asset]: yeah, that can be removed. it's a bug in the build system I believe

**Comment by jared-improbable** [comment]
@gkassabli If you rebase on the latest `master`, you will no longer need to check in the CoreSdk libraries.

**Comment by gkassabli** [comment]
@jared-improbable Right now we don't have worker packages for iOS that I can add to your changes, so I'm going to merge this PR as is for now.
I'll start working on adding packages and merging your changes into mobile branch right away

**Pull request merged**